app/lib/actions/katello/applicability/hosts/bulk_generate.rb

```diff
@@ -8,14 +8,20 @@ class BulkGenerate < Actions::EntryAction
           end
 
           def run
+            error = false
             input[:host_ids].each do |host_id|
               content_facet = ::Katello::Host::ContentFacet.find_by_host_id(host_id)
-              if content_facet.present?
-                content_facet.calculate_and_import_applicability
-              else
-                Rails.logger.warn(_("Content Facet for host with id %s is non-existent. Skipping applicability calculation.") % host_id)
-              end
+              # Catch errors and log them, but continue processing the rest of the hosts
+              content_facet.calculate_and_import_applicability
```

Comment:
Should we still keep the `present?` to not have to fail when host gets deleted before getting here?

Reply:
I'll put that back